nixos/modules/hardware/logiops/default.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.logiops;
+  renderedConfig =
+    (pkgs.formats.libconfig { }).generate "logid.cfg" cfg.settings;
+in
+{
+  options.services.logiops = {
+    enable = mkEnableOption "Logiops HID++ configuration";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.logiops;
+      defaultText = literalExpression "pkgs.logiops";
+      description = "The package to use for logiops";
+    };
+
+    settings = mkOption {
+      inherit (pkgs.formats.libconfig { }) type;
+      default = { };
+      example = {
+        devices = [{
+          name = "Wireless Mouse MX Master 3";
+
+          smartshift = {
+            on = true;
+            threshold = 20;
+          };
+
+          hiresscroll = {
+            hires = true;
+            invert = false;
+            target = false;
+          };
+
+          dpi = 1500;
+
+          buttons = [
+            {
+              cid = "0x53";
```

Comment:
My mistake. The literal `0x53` is necessary not the string `"0x53"`. At the moment this generates the error `[WARN] Line 16: cid must be a number, ignoring.`

Reply:
https://hyperrealm.github.io/libconfig/libconfig_manual.html#Integer-Values Hex is part of the spec.

Reply:
Applying your patch from the other thread resolves this. (also added a bit more)

```nix
nix-repl> :l .
nix-repl> :b (formats.libconfig {}).generate "test" example
https://i.ckie.dev/hPpbjIA.txt (browser default is tab width 8)